models/spike_generator.cpp

```diff
@@ -168,15 +167,29 @@ nest::spike_generator::Parameters_::assert_valid_spike_time_and_insert_( double
 }
 
 void
-nest::spike_generator::Parameters_::set( const DictionaryDatum& d,
-  State_& s,
+nest::spike_generator::SpikeParameters_::set( const DictionaryDatum& d,
+  SpikeState_& s,
   const Time& origin,
   const Time& now,
   Node* node )
 {
   bool flags_changed = updateValueParam< bool >( d, names::precise_times, precise_times_, node )
     or updateValueParam< bool >( d, names::shift_now_spikes, shift_now_spikes_, node )
     or updateValueParam< bool >( d, names::allow_offgrid_times, allow_offgrid_times_, node );
+
+  if ( d->known( names::allow_offgrid_times ) )
+  {
+    allow_offgrid_times_ = getValue< bool >( d->lookup( names::allow_offgrid_times ) );
+  }
+  if ( d->known( names::precise_times ) )
+  {
+    precise_times_ = getValue< bool >( d->lookup( names::precise_times ) );
+  }
+  if ( d->known( names::shift_now_spikes ) )
+  {
+    shift_now_spikes_ = getValue< bool >( d->lookup( names::shift_now_spikes ) );
+  }
+
```

Comment:
Why is this here? If I'm not completely mistaken, the block above already updates the values if they are known. This is what the `updateValueParam()` function is supposed to do. @hakonsbm, please correct me if I'm wrong.

Reply:
That's right, `updateValueParam()` will update the variable passed to it if the name is known, and return a `bool` stating if the variable was updated. There is therefore no need for these `if`-tests.

Reply:
solved